diff --git a/node_modules/image-size/dist/detector.cjs b/node_modules/image-size/dist/detector.cjs
index 0898608..aced8ae 100644
--- a/node_modules/image-size/dist/detector.cjs
+++ b/node_modules/image-size/dist/detector.cjs
@@ -171,7 +171,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
-      currentOffset = ispeBox.offset + ispeBox.size;
+      currentOffset = ispeBox.offset + (ispeBox.size > 0 ? ispeBox.size : 8);
     }
     if (images.length === 0) {
       throw new TypeError("Invalid HEIF, no sizes found");
@@ -253,7 +253,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize = getImageSize2(imageHeader[0]);
       images.push(imageSize);
-      imageOffset += imageHeader[1];
+      imageOffset += imageHeader[1] > 0 ? imageHeader[1] : 8;
     }
     if (images.length === 0) {
       throw new TypeError("Invalid ICNS, no sizes found");
@@ -488,7 +488,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
-    offset = jxlpBox.offset + jxlpBox.size;
+    offset = jxlpBox.offset + (jxlpBox.size > 0 ? jxlpBox.size : 8);
   }
   return partialStreams;
 }
diff --git a/node_modules/image-size/dist/detector.mjs b/node_modules/image-size/dist/detector.mjs
index 67afcab..0d4d52f 100644
--- a/node_modules/image-size/dist/detector.mjs
+++ b/node_modules/image-size/dist/detector.mjs
@@ -169,7 +169,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
-      currentOffset = ispeBox.offset + ispeBox.size;
+      currentOffset = ispeBox.offset + (ispeBox.size > 0 ? ispeBox.size : 8);
     }
     if (images.length === 0) {
       throw new TypeError("Invalid HEIF, no sizes found");
@@ -251,7 +251,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize = getImageSize2(imageHeader[0]);
       images.push(imageSize);
-      imageOffset += imageHeader[1];
+      imageOffset += imageHeader[1] > 0 ? imageHeader[1] : 8;
     }
     if (images.length === 0) {
       throw new TypeError("Invalid ICNS, no sizes found");
@@ -486,7 +486,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
-    offset = jxlpBox.offset + jxlpBox.size;
+    offset = jxlpBox.offset + (jxlpBox.size > 0 ? jxlpBox.size : 8);
   }
   return partialStreams;
 }
diff --git a/node_modules/image-size/dist/fromFile.cjs b/node_modules/image-size/dist/fromFile.cjs
index c226d3d..861e0be 100644
--- a/node_modules/image-size/dist/fromFile.cjs
+++ b/node_modules/image-size/dist/fromFile.cjs
@@ -197,7 +197,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
-      currentOffset = ispeBox.offset + ispeBox.size;
+      currentOffset = ispeBox.offset + (ispeBox.size > 0 ? ispeBox.size : 8);
     }
     if (images.length === 0) {
       throw new TypeError("Invalid HEIF, no sizes found");
@@ -279,7 +279,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize2 = getImageSize2(imageHeader[0]);
       images.push(imageSize2);
-      imageOffset += imageHeader[1];
+      imageOffset += imageHeader[1] > 0 ? imageHeader[1] : 8;
     }
     if (images.length === 0) {
       throw new TypeError("Invalid ICNS, no sizes found");
@@ -514,7 +514,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
-    offset = jxlpBox.offset + jxlpBox.size;
+    offset = jxlpBox.offset + (jxlpBox.size > 0 ? jxlpBox.size : 8);
   }
   return partialStreams;
 }
diff --git a/node_modules/image-size/dist/fromFile.mjs b/node_modules/image-size/dist/fromFile.mjs
index 4a3c444..38cd262 100644
--- a/node_modules/image-size/dist/fromFile.mjs
+++ b/node_modules/image-size/dist/fromFile.mjs
@@ -174,7 +174,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
-      currentOffset = ispeBox.offset + ispeBox.size;
+      currentOffset = ispeBox.offset + (ispeBox.size > 0 ? ispeBox.size : 8);
     }
     if (images.length === 0) {
       throw new TypeError("Invalid HEIF, no sizes found");
@@ -256,7 +256,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize2 = getImageSize2(imageHeader[0]);
       images.push(imageSize2);
-      imageOffset += imageHeader[1];
+      imageOffset += imageHeader[1] > 0 ? imageHeader[1] : 8;
     }
     if (images.length === 0) {
       throw new TypeError("Invalid ICNS, no sizes found");
@@ -491,7 +491,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
-    offset = jxlpBox.offset + jxlpBox.size;
+    offset = jxlpBox.offset + (jxlpBox.size > 0 ? jxlpBox.size : 8);
   }
   return partialStreams;
 }
diff --git a/node_modules/image-size/dist/index.cjs b/node_modules/image-size/dist/index.cjs
index bde0210..a850884 100644
--- a/node_modules/image-size/dist/index.cjs
+++ b/node_modules/image-size/dist/index.cjs
@@ -173,7 +173,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
-      currentOffset = ispeBox.offset + ispeBox.size;
+      currentOffset = ispeBox.offset + (ispeBox.size > 0 ? ispeBox.size : 8);
     }
     if (images.length === 0) {
       throw new TypeError("Invalid HEIF, no sizes found");
@@ -255,7 +255,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize2 = getImageSize2(imageHeader[0]);
       images.push(imageSize2);
-      imageOffset += imageHeader[1];
+      imageOffset += imageHeader[1] > 0 ? imageHeader[1] : 8;
     }
     if (images.length === 0) {
       throw new TypeError("Invalid ICNS, no sizes found");
@@ -490,7 +490,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
-    offset = jxlpBox.offset + jxlpBox.size;
+    offset = jxlpBox.offset + (jxlpBox.size > 0 ? jxlpBox.size : 8);
   }
   return partialStreams;
 }
diff --git a/node_modules/image-size/dist/index.mjs b/node_modules/image-size/dist/index.mjs
index 22c17af..485e761 100644
--- a/node_modules/image-size/dist/index.mjs
+++ b/node_modules/image-size/dist/index.mjs
@@ -169,7 +169,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
-      currentOffset = ispeBox.offset + ispeBox.size;
+      currentOffset = ispeBox.offset + (ispeBox.size > 0 ? ispeBox.size : 8);
     }
     if (images.length === 0) {
       throw new TypeError("Invalid HEIF, no sizes found");
@@ -251,7 +251,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize2 = getImageSize2(imageHeader[0]);
       images.push(imageSize2);
-      imageOffset += imageHeader[1];
+      imageOffset += imageHeader[1] > 0 ? imageHeader[1] : 8;
     }
     if (images.length === 0) {
       throw new TypeError("Invalid ICNS, no sizes found");
@@ -486,7 +486,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
-    offset = jxlpBox.offset + jxlpBox.size;
+    offset = jxlpBox.offset + (jxlpBox.size > 0 ? jxlpBox.size : 8);
   }
   return partialStreams;
 }
diff --git a/node_modules/image-size/dist/lookup.cjs b/node_modules/image-size/dist/lookup.cjs
index 9731ad8..815a730 100644
--- a/node_modules/image-size/dist/lookup.cjs
+++ b/node_modules/image-size/dist/lookup.cjs
@@ -171,7 +171,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
-      currentOffset = ispeBox.offset + ispeBox.size;
+      currentOffset = ispeBox.offset + (ispeBox.size > 0 ? ispeBox.size : 8);
     }
     if (images.length === 0) {
       throw new TypeError("Invalid HEIF, no sizes found");
@@ -253,7 +253,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize2 = getImageSize2(imageHeader[0]);
       images.push(imageSize2);
-      imageOffset += imageHeader[1];
+      imageOffset += imageHeader[1] > 0 ? imageHeader[1] : 8;
     }
     if (images.length === 0) {
       throw new TypeError("Invalid ICNS, no sizes found");
@@ -488,7 +488,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
-    offset = jxlpBox.offset + jxlpBox.size;
+    offset = jxlpBox.offset + (jxlpBox.size > 0 ? jxlpBox.size : 8);
   }
   return partialStreams;
 }
diff --git a/node_modules/image-size/dist/lookup.mjs b/node_modules/image-size/dist/lookup.mjs
index f787d1c..f580ed2 100644
--- a/node_modules/image-size/dist/lookup.mjs
+++ b/node_modules/image-size/dist/lookup.mjs
@@ -169,7 +169,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
-      currentOffset = ispeBox.offset + ispeBox.size;
+      currentOffset = ispeBox.offset + (ispeBox.size > 0 ? ispeBox.size : 8);
     }
     if (images.length === 0) {
       throw new TypeError("Invalid HEIF, no sizes found");
@@ -251,7 +251,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize2 = getImageSize2(imageHeader[0]);
       images.push(imageSize2);
-      imageOffset += imageHeader[1];
+      imageOffset += imageHeader[1] > 0 ? imageHeader[1] : 8;
     }
     if (images.length === 0) {
       throw new TypeError("Invalid ICNS, no sizes found");
@@ -486,7 +486,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
-    offset = jxlpBox.offset + jxlpBox.size;
+    offset = jxlpBox.offset + (jxlpBox.size > 0 ? jxlpBox.size : 8);
   }
   return partialStreams;
 }
diff --git a/node_modules/image-size/dist/types/heif.cjs b/node_modules/image-size/dist/types/heif.cjs
index 1f49078..cb49c56 100644
--- a/node_modules/image-size/dist/types/heif.cjs
+++ b/node_modules/image-size/dist/types/heif.cjs
@@ -69,7 +69,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
-      currentOffset = ispeBox.offset + ispeBox.size;
+      currentOffset = ispeBox.offset + (ispeBox.size > 0 ? ispeBox.size : 8);
     }
     if (images.length === 0) {
       throw new TypeError("Invalid HEIF, no sizes found");
diff --git a/node_modules/image-size/dist/types/heif.mjs b/node_modules/image-size/dist/types/heif.mjs
index 0133091..57f4d1f 100644
--- a/node_modules/image-size/dist/types/heif.mjs
+++ b/node_modules/image-size/dist/types/heif.mjs
@@ -67,7 +67,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
-      currentOffset = ispeBox.offset + ispeBox.size;
+      currentOffset = ispeBox.offset + (ispeBox.size > 0 ? ispeBox.size : 8);
     }
     if (images.length === 0) {
       throw new TypeError("Invalid HEIF, no sizes found");
diff --git a/node_modules/image-size/dist/types/icns.cjs b/node_modules/image-size/dist/types/icns.cjs
index 32c8d37..1c0690e 100644
--- a/node_modules/image-size/dist/types/icns.cjs
+++ b/node_modules/image-size/dist/types/icns.cjs
@@ -74,7 +74,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize = getImageSize(imageHeader[0]);
       images.push(imageSize);
-      imageOffset += imageHeader[1];
+      imageOffset += imageHeader[1] > 0 ? imageHeader[1] : 8;
     }
     if (images.length === 0) {
       throw new TypeError("Invalid ICNS, no sizes found");
diff --git a/node_modules/image-size/dist/types/icns.mjs b/node_modules/image-size/dist/types/icns.mjs
index 8710027..f3becc7 100644
--- a/node_modules/image-size/dist/types/icns.mjs
+++ b/node_modules/image-size/dist/types/icns.mjs
@@ -72,7 +72,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize = getImageSize(imageHeader[0]);
       images.push(imageSize);
-      imageOffset += imageHeader[1];
+      imageOffset += imageHeader[1] > 0 ? imageHeader[1] : 8;
     }
     if (images.length === 0) {
       throw new TypeError("Invalid ICNS, no sizes found");
diff --git a/node_modules/image-size/dist/types/index.cjs b/node_modules/image-size/dist/types/index.cjs
index 6949cd1..3fe762c 100644
--- a/node_modules/image-size/dist/types/index.cjs
+++ b/node_modules/image-size/dist/types/index.cjs
@@ -171,7 +171,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
-      currentOffset = ispeBox.offset + ispeBox.size;
+      currentOffset = ispeBox.offset + (ispeBox.size > 0 ? ispeBox.size : 8);
     }
     if (images.length === 0) {
       throw new TypeError("Invalid HEIF, no sizes found");
@@ -253,7 +253,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize = getImageSize2(imageHeader[0]);
       images.push(imageSize);
-      imageOffset += imageHeader[1];
+      imageOffset += imageHeader[1] > 0 ? imageHeader[1] : 8;
     }
     if (images.length === 0) {
       throw new TypeError("Invalid ICNS, no sizes found");
@@ -488,7 +488,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
-    offset = jxlpBox.offset + jxlpBox.size;
+    offset = jxlpBox.offset + (jxlpBox.size > 0 ? jxlpBox.size : 8);
   }
   return partialStreams;
 }
diff --git a/node_modules/image-size/dist/types/index.mjs b/node_modules/image-size/dist/types/index.mjs
index b026dda..d93db0b 100644
--- a/node_modules/image-size/dist/types/index.mjs
+++ b/node_modules/image-size/dist/types/index.mjs
@@ -169,7 +169,7 @@ var HEIF = {
         width = rawWidth - cropRight;
       }
       images.push({ height, width });
-      currentOffset = ispeBox.offset + ispeBox.size;
+      currentOffset = ispeBox.offset + (ispeBox.size > 0 ? ispeBox.size : 8);
     }
     if (images.length === 0) {
       throw new TypeError("Invalid HEIF, no sizes found");
@@ -251,7 +251,7 @@ var ICNS = {
       const imageHeader = readImageHeader(input, imageOffset);
       const imageSize = getImageSize2(imageHeader[0]);
       images.push(imageSize);
-      imageOffset += imageHeader[1];
+      imageOffset += imageHeader[1] > 0 ? imageHeader[1] : 8;
     }
     if (images.length === 0) {
       throw new TypeError("Invalid ICNS, no sizes found");
@@ -486,7 +486,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
-    offset = jxlpBox.offset + jxlpBox.size;
+    offset = jxlpBox.offset + (jxlpBox.size > 0 ? jxlpBox.size : 8);
   }
   return partialStreams;
 }
diff --git a/node_modules/image-size/dist/types/jxl.cjs b/node_modules/image-size/dist/types/jxl.cjs
index a12095b..93d5f01 100644
--- a/node_modules/image-size/dist/types/jxl.cjs
+++ b/node_modules/image-size/dist/types/jxl.cjs
@@ -119,7 +119,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
-    offset = jxlpBox.offset + jxlpBox.size;
+    offset = jxlpBox.offset + (jxlpBox.size > 0 ? jxlpBox.size : 8);
   }
   return partialStreams;
 }
diff --git a/node_modules/image-size/dist/types/jxl.mjs b/node_modules/image-size/dist/types/jxl.mjs
index 49b0c4c..24daee3 100644
--- a/node_modules/image-size/dist/types/jxl.mjs
+++ b/node_modules/image-size/dist/types/jxl.mjs
@@ -117,7 +117,7 @@ function extractPartialStreams(input) {
     partialStreams.push(
       input.slice(jxlpBox.offset + 12, jxlpBox.offset + jxlpBox.size)
     );
-    offset = jxlpBox.offset + jxlpBox.size;
+    offset = jxlpBox.offset + (jxlpBox.size > 0 ? jxlpBox.size : 8);
   }
   return partialStreams;
 }
